src/selectedtests/work_items/process_task_mapping_work_items.py

```diff
@@ -81,23 +93,56 @@ def _run_create_task_mappings(
     :param work_item: An instance of ProjectTestMappingWorkItem.
     :param after_date: The date at which to start analyzing commits of the project.
     """
-    source_re = re.compile(work_item.source_file_regex)
-    module_source_re = None
-    if work_item.module:
-        module_source_re = re.compile(work_item.module_source_file_regex)
-
-    mappings, _ = TaskMappings.create_task_mappings(
+    mappings, most_recent_version_analyzed = generate_task_mappings(
         evg_api,
         work_item.project,
         VersionLimit(stop_at_date=after_date),
-        source_re,
+        work_item.source_file_regex,
         module_name=work_item.module,
-        module_file_regex=module_source_re,
-        build_regex=work_item.build_variant_regex,
+        module_source_file_regex=work_item.module_source_file_regex,
+        build_variant_regex=work_item.build_variant_regex,
     )
-    transformed_mappings = mappings.transform()
-    if transformed_mappings:
-        mongo.task_mappings().insert_many(transformed_mappings)
+    _create_project_in_task_mappings_config(mongo, work_item, most_recent_version_analyzed)
+    if mappings:
+        mongo.task_mappings().insert_many(mappings)
+    else:
+        LOGGER.info("No task mappings generated")
     log.info("Finished task mapping work item processing")
 
     return True
+
+
+def _update_task_mappings_config(mongo, project, most_recent_version_analyzed):
+    mongo.task_mappings_project_config().update_one(
+        {"project": project},
+        {"$set": {"most_recent_version_analyzed": most_recent_version_analyzed}},
+    )
+
+
+def update_task_mappings_since_last_commit(evg_api: EvergreenApi, mongo: MongoWrapper):
+    """
+    Update task mappings that are being tracked in the task mappings project config collection.
+
+    :param evg_api: An instance of the evg_api client
+    :param mongo: An instance of MongoWrapper.
+    """
+    LOGGER.info("Updating task mappings")
+    project_cursor = mongo.task_mappings_project_config().find({})
+    for project_config in project_cursor:
```

Comment:
From a design perspective, it might make sense to create a `project_config` domain object that abstracts the project configs. That could remove a lot of the duplication from the `*_work_items` files and keep all the logic for tracking project configs in a single location.

Reply:
Added